src/storage/sql_table.cpp

```diff
@@ -168,16 +172,96 @@ void SqlTable::Scan(const terrier::common::ManagedPointer<transaction::Transacti
   }
 }
 
+// return false if it is null
+static bool PeekValue(const type::TransientValue &transient_val, byte *value_output) {
+  // Value output should be zero filled before calling
+  if (transient_val.Null()) {
+    // NullToSql(&expr) produces a NULL of expr's type.
+    return false;
+  }
+
+  switch (transient_val.Type()) {
+    case type::TypeId::BOOLEAN: {
+      auto val = type::TransientValuePeeker::PeekBoolean(transient_val);
+      memcpy(value_output, &val, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::TINYINT: {
+      auto val = type::TransientValuePeeker::PeekTinyInt(transient_val);
+      memcpy(value_output, &val, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::SMALLINT: {
+      auto val = type::TransientValuePeeker::PeekSmallInt(transient_val);
+      memcpy(value_output, &val, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::INTEGER: {
+      auto val = type::TransientValuePeeker::PeekInteger(transient_val);
+      memcpy(value_output, &val, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::BIGINT: {
+      auto val = type::TransientValuePeeker::PeekBigInt(transient_val);
+      memcpy(value_output, &val, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::DATE: {
+      // TODO(Schema-Change): find a way to handle it without codegen
```

Comment:
why this one needs to be handled differently if it is a constant value? Or is it referring to the case when the default value is from some function like Now()?

Reply:
you can specify a fixed date as default value . this is not for Now() which is a builtin